src/Control/Effect/Parse.hs

```diff
@@ -0,0 +1,29 @@
+{-# LANGUAGE ExistentialQuantification #-}
+module Control.Effect.Parse
+( -- * Parse effect
+  Parse(..)
+, parse
```

Comment:
By factoring this out into its own effect, stuff that needs to parse no longer needs most of the other junk that `Task` came with, and thus can be simplified overall.